src/amdinfer/workers/rocal.cpp

```diff
@@ -0,0 +1,438 @@
+              break;
+          case RocalOperation::CropResizeFixed:
+              width = node["dest_width"].GetUint();
+              height = node["dest_height"].GetUint();
+              is_output = node.HasMember("is_output") ? node["is_output"].GetBool() : false;
+              area = node["area"].GetFloat();
+              aspect_ratio = node["aspect_ratio"].GetFloat();
+              x_center_drift = node["x_center_drift"].GetFloat();
+              y_center_drift = node["y_center_drift"].GetFloat();
+              outputImage = rocalCropResizeFixed(handle_, inputImage, width, height, is_output, area, aspect_ratio, x_center_drift, y_center_drift);
+              break;
+          case RocalOperation::CropMirrorNormalize:
+              height = node["crop_height"].GetUint();
+              width = node["crop_width"].GetUint();
+              x = node["start_x"].GetFloat();
+              y = node["start_y"].GetFloat();
+              mean = parseVector(node["mean"]);
+              std_dev = parseVector(node["std_dev"]);
+              is_output = node.HasMember("is_output") ? node["is_output"].GetBool() : false;
+              mirror = node.HasMember("mirror") ? node["mirror"].GetBool() : false;
+              p_mirror = rocalCreateIntParameter(mirror);
+              outputImage = rocalCropMirrorNormalize(handle_, inputImage, height, width, x, y, mean, std_dev, is_output, p_mirror);
+              break;
+          case RocalOperation::CropFixed:
+              width = node["crop_width"].GetUint();
+              height = node["crop_height"].GetUint();
+              depth = node["crop_depth"].GetUint();
+              is_output = node.HasMember("is_output") ? node["is_output"].GetBool() : false;
+              x = node["crop_pos_x"].GetFloat();
+              y = node["crop_pos_y"].GetFloat();
+              z = node["crop_pos_z"].GetFloat();
+              outputImage = rocalCropFixed(handle_, inputImage, width, height, depth, is_output, x, y, z);
+              break;
+          case RocalOperation::ResizeCropMirrorFixed:
+              width = node["dest_width"].GetUint();
+              height = node["dest_height"].GetUint();
+              is_output = node.HasMember("is_output") ? node["is_output"].GetBool() : false;
+              y = node["crop_h"].GetUint();
+              x = node["crop_w"].GetUint();
+              mirror = node.HasMember("mirror") ? node["mirror"].GetBool() : false;
+              p_mirror = rocalCreateIntParameter(mirror);
+              outputImage = rocalResizeCropMirrorFixed(handle_, inputImage, width, height, is_output, y, x, p_mirror);
+              break;
+          default:
+              std::string errorMessage = "Unrecognized operation";
+              std::cerr << errorMessage << std::endl;
+              throw std::invalid_argument(errorMessage);
+      }
+      images.push_back(outputImage);
```

Comment:
The last image(tensor) used will be the input to the next node. 
All the images(tensors) need do be present until the `rocalRun()` function is called.
Therefore, I'm storing them in a vector since we don't know the number of augmentation nodes defined by the clients until runtime.

Reply:
Maybe I'm missing something but you're creating `std::vector<RocalTensor> tensors` as a local variable inside `deserialize()`, which is called in the `doInit()`. Isn't this vector deallocated by the time `rocalRun()` is called in the `doRun()`?